sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/AggregateDocumentQueryExecutionContext.java

```diff
@@ -62,23 +61,23 @@ public AggregateDocumentQueryExecutionContext(IDocumentQueryExecutionComponent<T
                     HashMap<String, String> headers = new HashMap<>();
                     List<ClientSideRequestStatistics> diagnosticsList = new ArrayList<>();
 
-                    for(FeedResponse<T> page : superList) {
+                    for(FeedResponse<Document> page : superList) {
```

Comment:
why we have to use Document type here? but not T? what are the required properties which might be missing in T?

Reply:
oh I think I got it, some properties are used in  RewrittenAggregateProjections

Reply:
Correct - some of the components require Document - aggregate, dcount and order by